src/query/storages/fuse/src/io/read/block/block_reader_native.rs

```diff
@@ -239,7 +244,10 @@ impl BlockReader {
                 }
             }
         }
-        Ok(DataBlock::new(entries, nums_rows))
+        match nums_rows {
+            Some(nums_rows) => Ok(DataBlock::new(entries, nums_rows)),
+            None => Ok(DataBlock::empty()),
```

Comment:
If the chunks are empty, we will push `default_val` to `entries`, in this case, `DataBlock` is not empty.

Reply:
fixed